source/upgrading/upgrade/upgrade-4.13.rst

```diff
@@ -0,0 +1,360 @@
+APT. This guide assumes you'll be using RPM and Yum (for Red Hat
+Enterprise Linux or CentOS), or APT and Debian packages (for Ubuntu).
+
+Create RPM or Debian packages (as appropriate) and a repository from
+the |release| source, or check the Apache CloudStack downloads page at
+http://cloudstack.apache.org/downloads.html
+for package repositories supplied by community members. You will need
+them for :ref:`ubuntu412` or :ref:`rhel412` and :ref:`kvm412` hosts upgrade.
+
+Instructions for creating packages from the CloudStack source are in the
+`CloudStack Installation Guide`_.
+
+Database Preparation
+--------------------
+
+Backup current database
+
+#. Stop your management server or servers. Run this on all management
+   server hosts:
+
+   .. parsed-literal::
+
+      $ sudo service cloudstack-management stop
+
+#. If you are running a usage server or usage servers, stop those as well:
+
+   .. parsed-literal::
+
+      $ sudo service cloudstack-usage stop
+
+#. Make a backup of your MySQL database. If you run into any issues or
+   need to roll back the upgrade, this will assist in debugging or
+   restoring your existing environment. You'll be prompted for your
+   password.
+
+   .. parsed-literal::
+
+      $ mysqldump -u root -p cloud > cloud-backup_`date '+%Y-%m-%d'`.sql
+      $ mysqldump -u root -p cloud_usage > cloud_usage-backup_`date '+%Y-%m-%d'`.sql
+
+#. **(KVM Only)** If primary storage of type local storage is in use, the
+   path for this storage needs to be verified to ensure it passes new
+   validation. Check local storage by querying the cloud.storage\_pool
+   table:
+
+   .. parsed-literal::
+
+      $ mysql -u cloud -p -e "select id,name,path from cloud.storage_pool where pool_type='Filesystem'"
+
+   If local storage paths are found to have a trailing forward slash,
+   remove it:
+
+   .. parsed-literal::
+
+      $ mysql -u cloud -p -e 'update cloud.storage_pool set path="/var/lib/libvirt/images" where path="/var/lib/libvirt/images/"';
+
+
+.. _ubuntu412:
+.. _apt-repo412:
+
+Management Server
+-----------------
+
+Ubuntu
+######
+
+If you are using Ubuntu, follow this procedure to upgrade your packages. If
+not, skip to step :ref:`rhel411`.
+
+.. note::
+   **Community Packages:** This section assumes you're using the community
+   supplied packages for CloudStack. If you've created your own packages and
+   APT repository, substitute your own URL for the ones used in these examples.
+
+The first order of business will be to change the sources list for
+each system with CloudStack packages. This means all management
+servers, and any hosts that have the KVM agent. (No changes should
+be necessary for hosts that are running VMware or Xen.)
+
+
+
+Start by opening ``/etc/apt/sources.list.d/cloudstack.list`` on
+any systems that have CloudStack packages installed.
+
+This file should have one line, which contains:
+
+.. parsed-literal::
+
+   deb http://download.cloudstack.org/ubuntu precise 4.12
```

Comment:
4.13?

Reply:
Nope. it is saying to change the value from this to another version. Fixed the use of precise to bionic (18.04).

Reply:
this is upgrade from 4.13, so afaik makes sense that 4.13 is already there, not 4.12

Reply:
fixed